docs/frontend-system/building-apps/08-migrating.md

```diff
@@ -0,0 +1,435 @@
+
+For example, the following sign-in page configuration:
+
+```tsx
+const app = createApp({
+  components: {
+    SignInPage: props => (
+      <SignInPage
+        {...props}
+        provider={{
+          id: 'github-auth-provider',
+          title: 'GitHub',
+          message: 'Sign in using GitHub',
+          apiRef: githubAuthApiRef,
+        }}
+      />
+    ),
+  },
+});
+```
+
+Can be converted to the following extension:
+
+```tsx
+const signInPage = createSignInPageExtension({
+  loader: async () => props =>
+    (
+      <SignInPage
+        {...props}
+        provider={{
+          id: 'github-auth-provider',
+          title: 'GitHub',
+          message: 'Sign in using GitHub',
+          apiRef: githubAuthApiRef,
+        }}
+      />
+    ),
+});
+```
+
+### `themes`
+
+Themes are now installed as extensions instead using `createThemeExtension`. See the section on [configuring themes](./index.md#TODO) for more information.
+
+For example, the following theme configuration:
+
+```tsx
+const app = createApp({
+  themes: [
+    {
+      id: 'light',
+      title: 'Light',
+      variant: 'light',
+      Provider: ({ children }) => (
+        <UnifiedThemeProvider theme={customLightTheme} children={children} />
+      ),
+    },
+  ],
+```
+
+Can be converted to the following extension:
+
+```tsx
+const lightTheme = createThemeExtension({
+  id: 'light',
+  title: 'Light Theme',
+  variant: 'light',
+  icon: <LightIcon />,
+  Provider: ({ children }) => (
+    <UnifiedThemeProvider theme={builtinThemes.light} children={children} />
+  ),
+});
+```
+
+### `configLoader`
+
+The config loader API has been slightly changed. Rather than returning a promise for an array of `AppConfig` objects, it should now return the `ConfigApi` directly.
+
+```ts
+const app = createApp({
+  async configLoader() {
+    const appConfigs = await loadAppConfigs();
+    // highlight-remove-next-line
+    return appConfigs;
+    // highlight-add-next-line
+    return { config: ConfigReader.fromConfigs(appConfigs) };
+  },
+});
+```
+
+### `bindRoutes`
+
+Route bindings can still be done using this option, but you now also have the ability to bind routes using static configuration instead. See the section on [binding routes](../architecture/07-routes.md#binding-external-route-references) for more information.
+
+Note that if you are binding routes from a legacy plugin that was converted using `convertLegacyApp`, you will need to use the `convertLegacyRouteRefs` and/or `convertLegacyRouteRef` to convert the routes to be compatible with the new system.
+
+For example, if both the `catalogPlugin` and `scaffolderPlugin` are legacy plugins, you can bind their routes like this:
+
+```ts
+const app = createApp({
+  features: convertLegacyApp(...),
+  bindRoutes({ bind }) {
+    bind(convertLegacyRouteRefs(catalogPlugin.createComponent), {
```

Comment:
Should this still be accessed through the `externalRoutes` prop or did something change?
```suggestion
    bind(convertLegacyRouteRefs(catalogPlugin.externalRoutes.createComponent), {
```

Reply:
Ah this was a copy-paste derp, should be just `.externalRoutes` here and `createComponent` instead of `registerApi` below. Fix'd